File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int mini=prices[0];
        int n=prices.size();
        int maxProfit=0;

        for(int i=1;i<n;i++){
            int profit=prices[i]-mini;
            maxProfit=max(maxProfit,profit);

            mini=min(mini,prices[i]);
        }


        return maxProfit;
    }
};

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int maxprofit=0;
        int mini=prices[0];

        for(int i=1;i<prices.length;i++){
            int profit=prices[i]-mini;

            maxprofit=Math.max(profit,maxprofit);
            mini=Math.min(mini,prices[i]);
        }

        return maxprofit;
    }
}